collectors/gtfs/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/artonge/go-gtfs"
+	"github.com/go-co-op/gocron"
+	"github.com/joho/godotenv"
+	"github.com/kiel-live/kiel-live/client"
+	"github.com/kiel-live/kiel-live/protocol"
+	log "github.com/sirupsen/logrus"
+)
+
+const IDPrefix = "gtfs-"
+
+func main() {
+	log.Infof("Kiel-Live GTFS collector version %s", "1.0.0") // TODO use proper version
+
+	err := godotenv.Load()
+	if err != nil {
+		log.Debug("No .env file found")
+	}
+
+	if os.Getenv("LOG") == "debug" {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	server := os.Getenv("COLLECTOR_SERVER")
+	if server == "" {
+		log.Fatalln("Please provide a server address for the collector with COLLECTOR_SERVER")
+	}
+
+	token := os.Getenv("COLLECTOR_TOKEN")
+	if token == "" {
+		log.Fatalln("Please provide a token for the collector with MANAGER_TOKEN")
+	}
+
+	c := client.NewClient(server, client.WithAuth("collector", token))
+	err = c.Connect()
+	if err != nil {
+		log.Fatalln(err)
+		return
+	}
+	defer func() {
+		err := c.Disconnect()
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	g, err := loadRemoteGTFS("https://github.com/lukashass/nok-gtfs/raw/main/adler.zip")
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	s := gocron.NewScheduler(time.UTC)
+	s.SetMaxConcurrentJobs(1, gocron.RescheduleMode)
+	_, err = s.Every(1).Minute().Do(func() {
+		if !c.IsConnected() {
+			return
+		}
+
+		for _, gtfsStop := range g.Stops {
+			// convert to protocol.Stop
+			stop := protocol.Stop{
+				ID:   IDPrefix + gtfsStop.ID,
+				Name: gtfsStop.Name,
+				Type: protocol.StopTypeFerryStop,
+				Location: protocol.Location{
+					Latitude:  int(gtfsStop.Latitude * 3600000),
+					Longitude: int(gtfsStop.Longitude * 3600000),
+				},
+				Alerts: []string{"Die Abfahrtszeiten können sich je nach Witterung oder Verkehrslage auf dem Nord-Ostsee-Kanal geringfügig verschieben. Die Verschiebung einer Fahrt dient der Sicherheit des Fahrbetriebes. Bei Ausfall der Fähre ist ein Busersatzverkehr eingerichtet."},
+			}
+			// for each trip remove stop times with the highest stop_sequence (last stop)
+			filteredStopTimes := make([]gtfs.StopTime, 0)
+			for _, stopTime := range g.StopsTimes {
```

Comment:
If I get this correctly this is `O(n*n)`? Couldn't we create a filtered index from the stop time first and assign that index to the corresponding stop after that?